docs/user/adding-new-metrics.md

```diff
@@ -1,97 +1,216 @@
+
+In this diagram, the ping "lifetimes" are represented as rectangles, but the moment the ping is "submitted" is represented by its right edge. The user changes the "turbo mode" setting from `false` to `true` at 3 seconds.
+  
+```mermaid
+gantt
+  axisFormat %S
+  title Metric lifetimes
+  dateFormat  s
+
+  section Metrics<br/>pings
+    Ping 1 :active, p1, 0, 2s
+    Ping 2 :active, p2, 2, 3s
+    Ping 3 :active, p3, 5, 10s
+    Ping 4 :active, p4, 15, 5s
+    Ping 5 :active, p5, 20, 5s
+
+  section Process
+	Run 1     :done, a1, 0, 10s
+	Run 2     :done, a2, 15, 10s
 
-Metric identifiers (the combination of a metric's category and name) must be unique across all metrics that are sent by an application.
+  section Interaction
+    ↓ Turns on turbo mode : 3
+
+  section Ping<br/>(change)
+	True      :3, 2s
+
+  section Ping<br/>(init and change)
+    False     :0, 2s
+    True      :3, 2s
+    True      :15, 5s
+
+  section Application<br/>(change)
+    True      :3, 12s  
+
+  section Application<br/>(init and change)
+    False     :0, 3s
+    True      :3, 12s
+    True      :15, 10s
+
+  section User<br/>(change)
+    True      :3, 22s
+```
+
+- **Ping (change)**: The lifetime is set to `ping` and the value is recorded into the metric when it changes. The value isn't included in Ping 1, because Glean doesn't know about it yet.  It is included in the first ping after being recorded (Ping 2), which causes it to be cleared and it is not included in any subsequent pings.
+
+- **Ping (init and change)**: The lifetime is set to `ping` and the value is recorded at application initialization and when it changes. The default value is included in Ping 1, and the changed value is included in Ping 2, which causes it to be cleared.  It therefore misses Ping 3, but when the application is started, it is recorded again and it is included in Ping 4.  However, this causes it to be cleared again and it is not in Ping 5.
+
+- **Application (change)**: The lifetime is set to `application` and the value is recorded into the metric with it changes. The value isn't included in Ping 1, because Glean doesn't know about it yet. After the value is changed, it is included in Pings 2 and 3, but then due to application restart it is cleared, so it is not included in any subsequent pings.
+
+- **Application (init and change)**: The lifetime is set to `application` and the value is recorded at application initialization and when it changes. The default value is included in Ping 1, and the changed value is included in Pings 2 and 3. Even though the application startup causes it to be cleared, it is set again, and all subsequent pings also have the value.
+
+- **User (change)**: The lifetime is set to `user` and the value is recorded when it changes. Since `user` lifetime metrics aren't cleared unless the user profile is reset, it is included in all subsequent pings.
```

Comment:
It's worth noting that this fails to capture if the value was flipped between two pings (i.e. flag is on, then off, then again on ). We should probably highlight the disadvantages of this option as well, otherwise we risk this seen as the holy grail of the solutions

Reply:
Good point -- missing flipping twice during a ping lifetime is a disadvantage of *every* approach, though.  It's definitely worth noting that.

Reply:
Oh, right :)

Reply:
The solution there is to cut a ping when the value changes, but that's a second example where the ping's schedule _is_ under the data owner's control.